src/System.Private.CoreLib/shared/System/Number.Formatting.cs

```diff
@@ -488,6 +490,7 @@ private static string FormatSingle(ref ValueStringBuilder sb, float value, ReadO
             char fmt = ParseFormatSpecifier(format, out int digits);
             int precision = FloatPrecision;
             NumberBuffer number = default;
+            number.kind = NumberBufferKind.Double;
```

Comment:
Would it make more sense for `NumberBufferKind.Double` to be renamed to `NumberBufferKind.FloatingPoint` or similar?  It feels odd to use `Double` when we are working with `float`/`Single`

Reply:
Possibly, but I would like to do that as a follow-up PR.

Currently we use `DoubleToNumber` for both `single` and `double`.... We should probably either rename that as well, or add an explicit code-path for `SingleToNumber`

Reply:
We are introducing the `enum` in this PR.  Why not give it a good name from the start?

Reply:
Because, until the the methods are split to handle `Single` explicitly, it is actually a `NumberBuffer` which contains a `Double` (`Single` is upcast in order to make the call) and so the name is more accurate as-is.

Reply:
Ah, got it. That makes sense.